### PR TITLE
Video insertion

### PR DESCRIPTION
**Pull request by SergioEstevao**
Implementation of #602 

cc @bummytime, can you please test it, specially on iOS 7 and then code review?

### CODE REVIEW

**Comment by bummytime** [comment]
@astralbodies has volunteered to test this PR on iOS7!

:raised_hands:

File: Assets/ZSSRichTextEditor.js [hunk 1]
```diff
@@ -31,6 +31,9 @@ ZSSEditor.currentSelection;
 // The current editing image
 ZSSEditor.currentEditingImage;
 
+// The current editing image
```
Comment (bummytime) [inline Assets/ZSSRichTextEditor.js]: Update the comment? :-)

File: Assets/ZSSRichTextEditor.js [hunk 2]
```diff
@@ -920,6 +923,316 @@ ZSSEditor.removeImage = function(imageNodeIdentifier) {
 };
 
 /**
+ *  @brief Inserts a video tag using the videoURL as source and posterURL as the image to show while video is loading.
```
Comment (bummytime) [inline Assets/ZSSRichTextEditor.js]: Another style nit: newline after 100 chars

File: Assets/ZSSRichTextEditor.js [hunk 3]
```diff
@@ -920,6 +923,316 @@ ZSSEditor.removeImage = function(imageNodeIdentifier) {
...
+ *
+ *  @param videoURL  the url of the video to present
+ *  @param posterURL the url of an image to show while the video is loading
+ *  @param alt       the alt descriptuon when the video is not supported.
```
Comment (bummytime) [inline Assets/ZSSRichTextEditor.js]: `s/descriptuon/description`

File: Assets/ZSSRichTextEditor.js [hunk 4]
```diff
@@ -1860,6 +2182,39 @@ ZSSField.prototype.sendImageTappedCallback = function( imageNode ) {
...
+    var arguments = ['id=' + encodeURIComponent( videoId ),
+                     'url=' + encodeURIComponent( videoNode.src )];
+    
+    var joinedArguments = arguments.join( defaultCallbackSeparator );
+    
+    var thisObj = this;
+    
+    // WORKAROUND: force the event to become sort of "after-tap" through setTimeout()
+    //
+    setTimeout(function() { thisObj.callback('callback-video-tap', joinedArguments);}, 500);
```
Comment (bummytime) [inline Assets/ZSSRichTextEditor.js]: @SergioEstevao what is this code working around? If it was not here, does a touch event exhibit strange behavior?

File: Classes/WPEditorView.m [hunk 5]
```diff
@@ -736,6 +752,85 @@ - (void)handleImageTappedCallback:(NSURL*)url
...
+ *	@param		url		The url with all the callback information.
+ */
+- (void)handleVideoFullScreenStartedCallback:(NSURL *)aURL
+{
+    NSParameterAssert([aURL isKindOfClass:[NSURL class]]);
+    [self saveSelection];
+    // FIXME: SergioEstevao 2015/03/25 - It looks there is a bug on iOS 8 that makes
+    // the keyboard not to be hidden when a video is made to run in full screen inside a webview.
+    // this workaround searches for the first responder and dismisses it
+    UIView *firstResponder = [self findFirstResponder:self];
```
Comment (bummytime) [inline Classes/WPEditorView.m]: :+1:

**Comment by bummytime** [comment]
@SergioEstevao I walked through all the code and it looks good – lots of clear comments and code. I would like to take a little bit more time to test on device and will report back here shortly.

**Comment by bummytime** [comment]
@SergioEstevao you may want to run `pod install` again in the demo app. Looks like there are some updates.

File: Assets/ZSSRichTextEditor.js (hunk 4 above)
Comment (SergioEstevao) [inline Assets/ZSSRichTextEditor.js]: Good question I copy this from the image code. But I will test it without.

**Comment by astralbodies** [comment]
Looks good on iOS 7.  My only comments:
* When previewing the video after its attached, tapping on full screen briefly overlays the keyboard dismissing. Maybe we can not dismiss the keyboard since the full screen hides the keyboard?
* When initially giving permission to photos, tapping No presents a white screen. Dismissing and going back in then shows the "no permission" screen. This may not be related to this PR.

**Comment by SergioEstevao** [comment]
@astralbodies, can you save a video on of the dismissal of the keyboard in iOS7. I added code to fix a bug in iOS8 that kept the keyboard in front of the video. Maybe I need to deactivate it on iOS7.

The second issue is not related to this PR it looks a bug on the UIImagePicker.

**Comment by bummytime** [comment]
@SergioEstevao just tested your latest stuff. Seems pretty solid. Were there any final todo's that you wanted to get done?

**Comment by bummytime** [comment]
Ok @SergioEstevao, just ran through your latest. Let's do this! :shipit:

**Comment by bummytime** [comment]
:boom:

**Pull request merged**